mbs_results/ratio_of_means.py

```diff
@@ -0,0 +1,388 @@
+    df : pd.DataFrame
+         Original dataframe.
+    target : str
+        Column name of values to be imputed.
+    period : str
+        Column name containing datetime information.
+    reference : str
+        Column name of unique Identifier.
+    strata : str
+        Column name containing strata information (sic).
+    auxiliary : str
+        Column name containing auxiliary information (sic).
+
+    Returns
+    -------
+    pd.DataFrame
+        Original dataframe with imputed values in the target column, and with
+        intermediate columns which were used for the imputation method.
+    """
+
+    # Saving args to dict, so we can pass same attributes to multiple functions
+    # These arguments are used from the majority of functions
+    # Remove df to allow pipe chain
+
+    default_columns = locals()
+
+    del default_columns["df"]
+
+    if default_columns["filters"] is not None:
+
+        df = flag_rows_to_ignore(df, default_columns["filters"])
+
+    # TODO: Is datetime needed? If so here is a good place to convert to datetime
+    df["date"] = pd.to_datetime(df["date"], format="%Y%m")
+
+    # TODO: Pre calculated links
+
+    df = (
+        df.pipe(wrap_flag_matched_pairs, **default_columns)
+        .pipe(wrap_shift_by_strata_period, **default_columns)
+        .pipe(wrap_calculate_imputation_link, **default_columns)
+        .pipe(
+            create_impute_flags,
+            **default_columns,
+            predictive_auxiliary="f_predictive_auxiliary"
+        )
+        # TODO: How we gonna set defaults?
+        .fillna(
+            {"f_link_question": 1.0, "b_link_question": 1.0, "construction_link": 1.0}
+        )
+        .pipe(generate_imputation_marker)
+        .pipe(wrap_get_cumulative_links, **default_columns)
+        .pipe(
+            create_and_merge_imputation_values,
+            **default_columns,
+            imputation_class="group",
+            marker="imputation_marker",
+            combined_imputation="imputed_value",
+            cumulative_forward_link="cumulative_f_link_question",
+            cumulative_backward_link="cumulative_b_link_question",
+            construction_link="construction_link",
+            imputation_types=("c", "fir", "bir", "fic")
+        )
+    )
+
+    df = df.reset_index(drop=True)
+
+    df["date"] = pd.to_numeric(df["date"].dt.strftime("%Y%m"), errors="coerce")
+
+    df = df.drop(
+        columns=[
+            "f_match",
+            "b_match",
+            "flag_construction_matches",
+            "r_flag",
+            "fir_flag",
+            "bir_flag",
+            "c_flag",
+            "fic_flag",
+            "missing_value",
+            "imputation_group",
+            "cumulative_f_link_question",
+            "cumulative_b_link_question",
+            "f_predictive_question",
+            "b_predictive_question",
+            "ignore_from_link",
+            "filtered_target",
+        ],
+        axis=1,
+        errors="ignore",
+    )
```

Comment:
You *could* extact this into its own function which can be called. It might be nice to switch this on and off incase we need to verify the methods or methogology needs this. Also potential argument of selecting the needed columns vs dropping un-needed, guess whichever is a shorter list

Reply:
That's a lovely idea, i love it.  Should we get feedback from users and add it to the backlog ?

Reply:
Relates to 415

Reply:
Added TODO for this